sirepo_bluesky/shadow_handler.py

```diff
@@ -73,24 +80,35 @@ def read_shadow_file(filename, histogram_bins=None):
     beam.load(filename)
 
     # 1=X spatial coordinate; 3=Z spatial coordinate
-    data_dict = beam.histo2(1, 3, nolost=1, nbins=histogram_bins)
-    data = data_dict["histogram"]
+    with open(os.devnull, "w") as devnull:
+        with contextlib.redirect_stdout(devnull):
+            data_dict = beam.histo2(1, 3, nolost=1, nbins=histogram_bins)
 
-    # This returns a list of N values (N=number of rays)
-    photon_energy_list = Shadow.ShadowTools.getshcol(filename, col=11)  # 11=Energy [eV]
+            # This returns a list of N values (N=number of rays)
+            photon_energy_list = Shadow.ShadowTools.getshcol(filename, col=11)  # 11=Energy [eV]
+
+    data = data_dict["histogram"]
     photon_energy = np.mean(photon_energy_list)
 
-    return {
+    # convert to um
+    horizontal_extent = 1e3 * np.array(data_dict["xrange"][:2])
+    vertical_extent = 1e3 * np.array(data_dict["yrange"][:2])
```

Comment:
Shadow always uses units of mm, so we convert to um which is more convenient for beam sizes (shows up nicer in the callback, instead of 0.000... etc)